feat(go): add temp file leak section to troubleshooting (#8376)
getsentry/sentry-docs: src/platforms/go/common/troubleshooting.mdx

This is common in Go because you can compile the binary, and then have no more need for the source code. Without access to the source code, however, we can’t map anything back, and we don’t support uploading Go source code with a release to stitch this data on.

You can see [in the Serverless section](/platforms/go/serverless/#source-context) how to bundle the source code with the binary, which applies to other forms of deployment as well.

## Avoiding Temporary File Leaks In File Upload Handlers

> Go SDK middleware has to modify the request context in order to function correctly. This results in a shallow copy of the request being passed to upstream handlers. Temporary files created from calling `ParseMultipartForm` on this request will not automatically be cleaned up. It's important ensure these files are cleaned up by calling `request.Multipart.RemoveAll()` after.

Reviewer: ```suggestion
Sentry's Go SDK middleware has to modify the request context in order to function correctly. This results in a shallow copy of the request being passed to upstream handlers. This means that temporary files created from calling `ParseMultipartForm` on this request won't automatically be cleaned up. It's important to clean these files up by calling `request.Multipart.RemoveAll()` after.
```
